dwave_networkx/drawing/qubit_layout.py

```diff
@@ -468,34 +526,270 @@ def draw_yield(G, layout, perfect_graph, unused_color=(0.9,0.9,0.9,1.0),
+    else:
+        faults_node_color = [fault_color for v in faults_nodelist]
+        faults_edge_color = [fault_color for e in faults_edgelist]
+        incident_edge_color = [incident_fault_color for e in incident_edgelist]
+
+        # Draw edges first, in the order (unused, incident, faults)
+        if unused_color is not None:
+            unused_edge_color = [unused_color for e in G.edges()]
+            nx.draw_networkx_edges(G, layout, edge_color=unused_edge_color, **kwargs)
+        nx.draw_networkx_edges(perfect_graph, layout, incident_edgelist, style=fault_style, edge_color=incident_edge_color, **kwargs)
+        nx.draw_networkx_edges(perfect_graph, layout, faults_edgelist, style=fault_style, edge_color=faults_edge_color, **kwargs)
+
+        # Draw nodes second, in the order (unused, faults)
+        if unused_color is not None:
+            unused_node_color = [unused_color for e in G]
+            nx.draw_networkx_nodes(G, layout, node_color = unused_node_color, **kwargs)
+        nx.draw_networkx_nodes(perfect_graph, layout, faults_nodelist, node_shape=fault_shape, node_color=faults_node_color, **kwargs)
+
+
+def normalize_size_and_aspect(scale, node_scale, kwargs):
+    """Sets default values for the "ax", "node_size" and "width" keys.
+    
+    If the `"ax"` is not set, then we create an axis from the current
+    `matplotlib` figure (:func:`matplotlib.pyplot.gcf()`).  Then, the the axis
+    (either the one we created, or the pre-existing one) is set to have an
+    aspect ratio of 1 and the drawing of axes is turned off.
+    
+    Then, if `"node_size"` and `"width"` are not set, we compute default values
+    for them based on the scale of the current figure and the scale parameters
+    as appropriate.
+    
+    Note, if `kwargs["line_plot"]` is True, then the interpretation of the 
+    `"width"` and `"node_size"` parameters determine the area of edge-circles
+    and line widths (used for both nodes and edges) respectively.  See 
+    :func:`dwave_networkx.qubit_layout.draw_lineplot` for more information.
+
+    Parameters
+    ----------
+    scale : float
+        A geometric size of a figure to be drawn.  This is roughly the number
+        of qubits in any given row or column of the graph.
+        
+    node_scale : float
+        A magic number used to scale node circles -- ignored and replaced with
+        the number 50 when `kwargs["line_plot"]` is True.
+    
+    kwargs : dict
+        A dictionary to populate with default values.    
+    """
+    ax = kwargs.get('ax')
+    if ax is None:
+        try:
+            import matplotlib.pyplot as plt
+        except ImportError:
+            raise ImportError("Matplotlib required for graph drawing")
+        cf = plt.gcf()
+    else:
+        cf = ax.get_figure()
+    cf.set_facecolor("w")
+    if ax is None:
+        if cf._axstack() is None:
```

Comment:
@boothby, I started this morning but am running into broken functionality:
 ```
>>> import dwave_networkx as dnx
>>> G = dnx.chimera_graph(1)
>>> dnx.draw_chimera(G)
```
fails on ``TypeError: '_AxesStack' object is not callable``. 
From what I see, this line is not trigger by ``dnx.chimera_layout(G)`` but will be when that method has `kwargs`, which it will have when called through ``chimera_graph()``.

Also, I see that, 
```
>>> import matplotlib.pyplot as plt
>>> f = plt.figure()   # equivalent to cf = plt.gcf() in line 607
>>> f.__axstack()
```
fails on ``AttributeError: 'Figure' object has no attribute '__axstack'``

My venv uses ``matplotlib`` version 3.6.3. Is ``__axstack()`` supported in some previous version? This repo has no version requirement on matplotlib (or requirement that it be installed at all)

Reply:
Identical to https://github.com/networkx/networkx/pull/5937; this has an easy fix (thanks, git-blame!)